engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -964,7 +965,16 @@ else if (allocatedContainers.isEmpty() && numRequestedContainers == 0 && dnmgr.c
         appDone = true;
       }
 
-      LOG.debug("Current application state: loop=" + loopCounter + ", appDone=" + appDone + ", total=" + numTotalContainers + ", requested=" + numRequestedContainers + ", released=" + numReleasedContainers + ", completed=" + numCompletedContainers + ", failed=" + numFailedContainers + ", currentAllocated=" + allocatedContainers.size());
+      if (!StringUtils.isEmpty(dnmgr.shutdownDiagnosticsMessage)) {
```

Comment:
I don't think so. In this case we had to ask the user to reproduce the issue after enabling DEBUG and a message like this would have eliminated that need. To prevent such things I would like to have such INFO messages in the log (which here only happens when shutdown is requested).